connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -763,6 +789,30 @@ private boolean startTask(ConnectorTaskId taskId) {
         );
     }
 
+    private Callable<Void> getTaskCallable(final ConnectorTaskId taskId, final TaskStatus.State targetState) {
+        return new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                switch (targetState) {
+                    case RUNNING:
+                        try {
+                            startTask(taskId);
```

Comment:
Ah, I see. So yes, the primary win is for shutdown. Because we already push `configure()` and `start()` into the executor thread, you are right, we might not benefit much in the case of starting connectors. However, note that we do invoke connector code (technically the connector can override `initialize()`) and we also invoke some other code which, for some implementations, may not behave as you would like. In particular, we configure the converters, which it wouldn't be crazy to think might need to make a network connection that could cause startup to block or take some time (e.g. although our Avro converter doesn't happen to do this, it wouldn't be insane to start the HTTP request as part of initial setup).